central/alert/datastore/internal/store/postgres/store.go

```diff
@@ -179,45 +218,6 @@ func copyFromAlerts(ctx context.Context, s pgSearch.Deleter, tx *postgres.Tx, ob
 	// Which is essentially the desired behaviour of an upsert.
 	deletes := make([]string, 0, batchSize)
 
-	copyCols := []string{
```

Comment:
Looks like this is indeed an optimisation but can you verify it with a benchmark. We should see less allocs on copy from calls
```
central/alert/datastore/internal/store/postgres/store.go:182:22: []string{...} escapes to heap
```

Reply:
I added coverage for different sizes of UpsertMany batches in #17636
I think the one that would best highlight things would be the one with size 128 (the smallest above the batching threshold).